packages/app-store/googlecalendar/tests/google-calendar.e2e.ts

```diff
@@ -0,0 +1,215 @@
+import { expect } from "@playwright/test";
+import type { Page } from "@playwright/test";
+
+import dayjs from "@calcom/dayjs";
+import { APP_CREDENTIAL_SHARING_ENABLED } from "@calcom/lib/constants";
+import prisma from "@calcom/prisma";
+import type { Prisma } from "@calcom/prisma/client";
+import { test } from "@calcom/web/playwright/lib/fixtures";
+import { selectSecondAvailableTimeSlotNextMonth } from "@calcom/web/playwright/lib/testUtils";
+
+import metadata from "../_metadata";
+import GoogleCalendarService from "../lib/CalendarService";
+import { createBookingAndFetchGCalEvent, deleteBookingAndEvent, assertValueExists } from "./testUtils";
+
+test.describe("Google Calendar", async () => {
+  test.describe("Test using the primary calendar", async () => {
+    let qaUsername: string;
+    let qaGCalCredential: Prisma.CredentialGetPayload<{ select: { id: true } }>;
+    test.beforeAll(async () => {
+      let runIntegrationTest = false;
+
+      test.skip(!!APP_CREDENTIAL_SHARING_ENABLED, "Credential sharing enabled");
+
+      if (process.env.E2E_TEST_CALCOM_QA_EMAIL && process.env.E2E_TEST_CALCOM_QA_PASSWORD) {
+        qaGCalCredential = await prisma.credential.findFirstOrThrow({
+          where: {
+            user: {
+              email: process.env.E2E_TEST_CALCOM_QA_EMAIL,
+            },
+            type: metadata.type,
+          },
+          select: {
+            id: true,
+          },
+        });
+
+        const qaUserQuery = await prisma.user.findFirstOrThrow({
+          where: {
+            email: process.env.E2E_TEST_CALCOM_QA_EMAIL,
+          },
+          select: {
+            username: true,
+          },
+        });
+
+        assertValueExists(qaUserQuery.username, "qaUsername");
+        qaUsername = qaUserQuery.username;
+
+        if (qaGCalCredential && qaUsername) runIntegrationTest = true;
+      }
+
+      test.skip(!runIntegrationTest, "QA user not found");
```

Comment:
Atleast on CI we should error instead of skipping the tests. We could accidentally endup not running these tests there.

We should probably set  GOOGLE_CALENDAR_E2E_ENABLED=1 and error here if that is set. We would also need to set that in CI. It would allow this test to fail(instead of being skipped) in CI for us if we accidentally configure something wrong(i.e. QA user isn't found)

It would also allow tests to keep on passing for our self hosters till they opt in to these tests.